src/common/utils.h

```diff
@@ -983,9 +986,21 @@ inline bool AlignedMemAlloc(void** ptr, size_t size, size_t alignment) {
   if (*ptr == nullptr)
     return false;
 #else
-  int res = posix_memalign(ptr, alignment, size);
-  if (res != 0)
+  int res                       = posix_memalign(ptr, alignment, size);
+#if __linux__
+  constexpr size_t gHugePage2MB = 1 << 21;
+  if (size >= gHugePage2MB) {
+    // TODO(mozga-intel): Enable MacOS Huge Pages if desired
+    res = posix_memalign(ptr, gHugePage2MB, size);
+    if (res == 0) {
```

Comment:
on linux the `posix_memalign` would be called twice. intentional?

Reply:
Good catch! It should be called once.